sasmodels/sesans.py

```diff
@@ -58,23 +58,24 @@ def apply(self, Iq):
 
     def _set_hankel(self, SElength, lam, zaccept, Rmax):
         # type: (np.ndarray, float, float) -> None
-        # Force float32 arrays, otherwise run into memory problems on some machines
-        SElength = np.asarray(SElength, dtype='float32')
-
-        #Rmax = #value in text box somewhere in FitPage?
+        SElength = np.asarray(SElength)
         q_max = 2*pi / (SElength[1] - SElength[0])
         q_min = 0.1 * 2*pi / (np.size(SElength) * SElength[-1])
-        q = np.arange(q_min, q_max, q_min, dtype='float32')
-        dq = q_min
+        q = np.exp(np.arange(np.log(q_min), np.log(q_max), np.log(1.0003)))
```

Comment:
The hard-coded constant `1.0003` is suspicious.  Can you make it a default parameter in init?

Also, not sure how length of `q` various with the choice of `SElength`.  Can some experiments require dramatically larger number of q points to calculate?  Maybe use `np.logspace(np.log10(q_min), np.log10(q_max), C*SElength.size)` for `C` some user control parameter defaulting to 10, say, so that `q` length is proportional to `SElength`.

Reply:
I've gone ahead and made the constant a default parameter on the init. That's obviously the correct choice.

There was a conscious decision to try and make the length of `q` dependent only on the log of the length of `SElength`.  

    n = (log(10) + log(N) + log(zₓ/δz))/l
 where

    n = length of q
    N = length of SElength
    zₓ = maximum spin echo length
    δz = spacing between data points
    l = log(log_spacing)

You can then use the approximation that zₓ/δz ≈ N to get

   n ≈ (log(10) + 2 log(N))/l

Granted, there's a huge constant factor on this log scaling (around 7700 points for the case of N=1), but we can now handle millions of data points with only a 6x slowdown.